From 2cb53b22419fa33f7a85efda0c940116f5c60ea5 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 31 Jan 2026 15:31:59 +0000
Subject: [PATCH] tests/test-ipv4-lo.sh, tests/test-ipv6-lo.sh: Remove
 pick_unused_port

Remove use of pick_unused_port, and greatly simply these tests by
using --run and nbdinfo.

(cherry picked from commit e1ee17022c63d90dd7469c66ca4980e489fa3f95)
---
 tests/test-ipv4-lo.sh | 35 ++++-------------------------------
 tests/test-ipv6-lo.sh | 35 ++++-------------------------------
 2 files changed, 8 insertions(+), 62 deletions(-)

diff --git a/tests/test-ipv4-lo.sh b/tests/test-ipv4-lo.sh
index e0e1878a..9b6e5c42 100755
--- a/tests/test-ipv4-lo.sh
+++ b/tests/test-ipv4-lo.sh
@@ -38,36 +38,9 @@ set -e
 set -x
 set -u
 
-# Cannot use kill pidfile below to test if the process is running on
-# Windows.
-if is_windows; then
-    echo "$0: this test needs to be revised to work on Windows"
-    exit 77
-fi
-
-requires ip -V
-requires "$QEMU_IMG" --version
-requires "$QEMU_IMG" info --image-opts driver=file,filename=functions.sh
-
-rm -f ipv4lo.pid ipv4lo.out
-cleanup_fn rm -f ipv4lo.pid ipv4lo.out
-
-# Find an unused port to listen on.
-pick_unused_port
+requires_run
+requires_nbdinfo
 
 # By default nbdkit will listen on all available interfaces, ie.
-# IPv4 and IPv6.
-start_nbdkit -P ipv4lo.pid -p $port example1
-pid="$(cat ipv4lo.pid)"
-
-# Check the process exists.
-kill -s 0 $pid
-
-# Check we can connect over the IPv4 loopback interface.
-ipv4_lo="$(ip -o -4 addr show scope host)"
-if test -n "$ipv4_lo"; then
-    "$QEMU_IMG" info --output=json \
-        --image-opts "file.driver=nbd,file.host=127.0.0.1,file.port=$port" > ipv4lo.out
-    cat ipv4lo.out
-    grep -sq '"virtual-size": *104857600\b' ipv4lo.out
-fi
+# IPv4 and IPv6, so use -4 to force IPv4 only.
+nbdkit -v -4 --port=0 example1 --run 'nbdinfo nbd://127.0.0.1:$port/'
diff --git a/tests/test-ipv6-lo.sh b/tests/test-ipv6-lo.sh
index d30ae205..ea17bafc 100755
--- a/tests/test-ipv6-lo.sh
+++ b/tests/test-ipv6-lo.sh
@@ -38,37 +38,10 @@ set -e
 set -x
 set -u
 
-# Cannot use kill pidfile below to test if the process is running on
-# Windows.
-if is_windows; then
-    echo "$0: this test needs to be revised to work on Windows"
-    exit 77
-fi
-
-requires ip -V
-requires "$QEMU_IMG" --version
-requires "$QEMU_IMG" info --image-opts driver=file,filename=functions.sh
+requires_run
+requires_nbdinfo
 requires_ipv6_loopback
 
-rm -f ipv6lo.pid ipv6lo.out
-cleanup_fn rm -f ipv6lo.pid ipv6lo.out
-
-# Find an unused port to listen on.
-pick_unused_port
-
 # By default nbdkit will listen on all available interfaces, ie.
-# IPv4 and IPv6.
-start_nbdkit -P ipv6lo.pid -p $port example1
-pid="$(cat ipv6lo.pid)"
-
-# Check the process exists.
-kill -s 0 $pid
-
-# Check we can connect over the IPv6 loopback interface.
-ipv6_lo="$(ip -o -6 addr show scope host)"
-if test -n "$ipv6_lo"; then
-    "$QEMU_IMG" info --output=json \
-        --image-opts "file.driver=nbd,file.host=::1,file.port=$port" > ipv6lo.out
-    cat ipv6lo.out
-    grep -sq '"virtual-size": *104857600\b' ipv6lo.out
-fi
+# IPv4 and IPv6, so use -6 to force IPv6 only.
+nbdkit -v -6 --port=0 example1 --run 'nbdinfo nbd://[::1]:$port/'
-- 
2.47.3

